### PR TITLE
#166840910 Setup unit testing with mocha and chai

### PR DESCRIPTION
**Pull request by tohbay**
#### What does this PR do?
Setup unit testing with mocha and chai

#### Description of Task to be completed?
- ensure all implemented functionality is tested henceforth
- installed mocha, chai, chai-http, @babel/register, @babel/polyfill as devDependencies

#### How should this be manually tested?
The following steps can be used:
1) Clone the repository 
2) Cd into the project folder 
3) Run `npm install` to install all project dependencies. 
4) On the terminal run `npm test`
5) There should be a response showing that 2 tests are passing

#### Any background context you want to provide?
- Ensure that node and npm are installed on your machine

#### What are the relevant pivotal tracker stories?
Chore 166840910: https://www.pivotaltracker.com/story/show/166840910

#### Screenshots (if appropriate)
image

### CODE REVIEW

**Comment by Onyimatics** [comment]
Looks good to me

**Comment by kechyy** [comment]
Great Job Tobechukwu. Looks good to me but I feel you can checkout the way Onyinye linked her pivotal tracker story, I think that should be the normal convention to follow. It's quite short and linkable too. However, well done.

**Comment by tohbay** [comment]
> Great Job Tobechukwu. Looks good to me but I feel you can check out the way Onyinye **`linked her pivotal tracker story`**, I think that should be the normal convention to follow. It's quite **`short and linkable`** too. However, well done.

Thank you Nkechi for the feedback. I have checked Onyinye's task on the PT Board and I would love you specifically point me **(a screenshot if possible)** to the normal convention you mentioned above  **(areas I have highlighted in this comment)** so I can learn and improve on it.

Okay, sorry for not making it specific enough. I feel the convention goes this way: 
[#166840913](https://www.pivotaltracker.com/story/show/166840913). You can edit to see how it goes or Let me know if you would want me to do it. Thank you.

**Comment by tohbay** [comment]
Oh, I think this is what Nkechi was referring to in the details of my Pull Request:

image

This aspect of the Pull Request is just me putting down in full description of the task done. It was one of the lessons I learned from my Bootcamp Learning Facilitator. I was not allowed to just put a PT Board link without having details of what the link was pointing to.
Hence, I had to implement it here going forward and at the same time hoping for an update on this from our Learning Facilitators or Technical Team Leads.

**Comment by codexempire** [comment]
Looks good to me

File: src/test/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        done();
+      });
+  });
+  it('It should return a message on successful page load', done => {
+    const appUrl = '/';
+    chai
+      .request(app)
+      .get(appUrl)
+      .end((error, response) => {
+        expect(response.body).to.be.a('object');
```
Comment (segunolalive) [inline src/test/index.spec.js]: `response.body` is always an `object`. This assertion doesn't actually prove that the `api` works as expected.

File: src/test/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../index';
+
+chai.use(chaiHttp);
+
+describe('Testing home endpoint', () => {
+  it('It should return status of 200 on successful page load', done => {
```
Comment (segunolalive) [inline src/test/index.spec.js]: Are we loading a page?

**Review by segunolalive — CHANGES_REQUESTED** [review]
Good job 👍 , @tobechukwuobitube.

Take a look at the comments

File: src/test/index.spec.js (hunk 2 above)
Comment (tohbay) [reply src/test/index.spec.js]: No we are not loading a page. The test was just to test the functionality of the set up.

File: src/test/index.spec.js (hunk 1 above)
Comment (tohbay) [reply src/test/index.spec.js]: I agree with you. 
That assertion was just to ensure that the unit test was properly set up.
In some use cases, an assertion like that can be made in order to ensure that outputs meet the required expectation and acceptance criteria.

Comment (segunolalive) [reply src/test/index.spec.js]: the second assertion is useful. The first is not.

Remove it or change it to something some more explicit, like the properties expected on the response body.

File: src/test/index.spec.js (hunk 2 above)
Comment (segunolalive) [reply src/test/index.spec.js]: your test description should clearly explain what's being tested.

File: src/test/index.spec.js (hunk 1 above)
Comment (tohbay) [reply src/test/index.spec.js]: I've made the necessary changes to the most recent comment raised about the first assertion.

Comment (tohbay) [reply src/test/index.spec.js]: I have made changes to the comments you raised on the first assertion.

**Pull request merged**